web/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,434 @@
+            </svg>
+          </button>
+        </span>
+      </td>
+    </tr>
+  );
+}
+
+export default function Table({ data, currentPage, rowCountPerPage, visibleColumns, hideColumn, showHistogram, setShowQR, setQRURL, filters, setFilters, filterPopupRef, showLineNumbers }: TableProps) {
+  const [sortConfig, setSortConfig] = useState<{ key: keyof TableRowProps; direction: 'ascending' | 'descending' } | null>(null);
+  const [filterColumn, setFilterColumn] = useState<string | null>(null);
+  const [showFilterPopup, setShowFilterPopup] = useState<boolean>(false);
+  const [anchorRefs, setAnchorRefs] = useState<{ [key: string]: HTMLDivElement | null }>({});
+  const [filterType, setFilterType] = useState('')
+
+
+  const rowsPerPage = rowCountPerPage;
+  const indexOfLastRow = currentPage * rowsPerPage;
+  const indexOfFirstRow = indexOfLastRow - rowsPerPage;
+
+  const processData = React.useMemo(() => {
+    let processedData = [...data];
+
+    if (Object.keys(filters).length > 0) {
+      processedData = processedData.filter(row => {
+        return Object.keys(filters).every(column => {
+
+          
+        const columnIndex = config.columns.indexOf(column);
+        const cellValue = row[columnIndex]?.toString() || '';
+
+
+          if (config.displayed_numeric_columns.includes(column)) {
+
+            const [minValue, maxValue] = filters[column].split(',').map(Number);
+
+            if (!minValue || !maxValue) return data;
+
+
+            const numericCellValue = parseFloat(cellValue);
+
+            return numericCellValue >= minValue && numericCellValue <= maxValue;
+
+          } else if (config.unique_column_values[column]) {
+
+            if (config.unique_column_values[column] > 10) {
+              const filterValue = filters[column];
+              return cellValue.includes(filterValue);
+            } else if (config.unique_column_values[column] <= 10) {
+              var choosenValues: any = []
+
+              filters[column].split(',').map((value: any) => {
+                if (value == '') return;
+                choosenValues.push(value)
+              })
+
+              return choosenValues.some((value: string) => cellValue.includes(value));
+            }
+          } else {
+            const filterValue = filters[column];
+            return cellValue.includes(filterValue);
+          }
+        });
+      });
+    }
+    
+    if (sortConfig) {
+      processedData.sort((a, b) => {
+        const columnIndex = config.columns.indexOf(sortConfig.key);
+        if (a[columnIndex] < b[columnIndex]) {
+          return sortConfig.direction === 'ascending' ? -1 : 1;
+        }
+        if (a[columnIndex] > b[columnIndex]) {
+          return sortConfig.direction === 'ascending' ? 1 : -1;
+        }
+        return 0;
+      });
+    }
+
+    return processedData;
+  }, [data, filters, sortConfig, config.columns]);
```

Comment:
This code seems very nested and unclear to me. I would suggest breaking down the logic into smaller, reusable functions to enhance readability and maintainability. This will make it easier to manage and debug in the future. A few comments could also help to understand the logic.